app/services/teacher_training_api/create_course.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module TeacherTrainingApi
+  class CreateCourse
+    include ServicePattern
+
+    def initialize(provider:, course:)
+      @provider = provider
+      @attrs = course["attributes"]
+    end
+
+    def call
+      course.update!(name: attrs["name"])
+    end
+
+  private
+
+    attr_reader :provider, :attrs
+
+    def course
+      @course ||= provider.courses.find_or_initialize_by(code: attrs["code"])
```

Comment:
![image](https://user-images.githubusercontent.com/470137/111134110-3319e380-8573-11eb-917c-06e9bd7d8869.png)

@gpeng 

Do we need to only capture the courses that's is a non draft?

Reply:
Sounds to me like we probably shouldn't be creating courses that are still in draft... but I'll defer to @gpeng !

Reply:
I don't think we should 'import' draft courses but I think once they're published we should keep the other states. Ultimately this will become a list that the user selects from which then populates the course details fields for the trainee so if there are extra courses in that list e.g. withdrawn or something like that then that's not a problem.

Reply:
cool drop draft :+1:

Reply:
@defong @gpeng Thanks guys! Ready for re-review, `CreateCourse` now skips over courses with state "draft".